src/lib/bootloader/grub2.rb

```diff
@@ -121,6 +151,14 @@ def packages
 
   private
 
+    def replace_with(other)
+      stage1.clear_devices
```

Comment:
Would it make sense to have some method `#Stage1#replace` which implements this logic?